balancer/roundrobin/roundrobin.go

```diff
@@ -47,12 +48,19 @@ type rrPickerBuilder struct{}
 
 func (*rrPickerBuilder) Build(readySCs map[resolver.Address]balancer.SubConn) balancer.Picker {
 	grpclog.Infof("roundrobinPicker: newPicker called with readySCs: %v", readySCs)
+	if len(readySCs) == 0 {
+		return base.NewErrPicker(balancer.ErrNoSubConnAvailable)
+	}
 	var scs []balancer.SubConn
 	for _, sc := range readySCs {
```

Comment:
So go is iterating through the channels in a shuffled order?

Reply:
Effectively, yes.  They are ordered by how they happen to be in the readySCs map, keyed by the address (as a string, plus a "type" and a pointer to optional user metadata).  This is also not right, but I do think randomizing here is better than not randomizing, so we should merge this PR for now and file an issue to improve further.

Reply:
Yes, I wasn't wanting to block this PR.